src/cross_section.F90

```diff
@@ -526,6 +572,185 @@ pure function find_energy_index(mat, E) result(i)
   end function find_energy_index
 
 !===============================================================================
+! MULTIPOLE_EVAL_ALLOCATE allocates fixed-length arrays that vary based on
+! what nuclides are loaded into the problem
+!===============================================================================
+
+  subroutine multipole_eval_allocate()
+    allocate(sigT_factor(max_L))
+    allocate(twophi(max_L))
+    allocate(broadened_polynomials(max_poly))
+
+    mp_already_alloc = .TRUE.
+  end subroutine
+
+!===============================================================================
+! MULTIPOLE_EVAL evaluates the windowed multipole equations for cross
+! sections in the resolved resonance regions
+!===============================================================================
+
+  subroutine multipole_eval(multipole, Emev, sqrtkT, sigT, sigA, sigF)
+    type(MultipoleArray), intent(in) :: multipole                    !< The windowed multipole object to process.
+    real(8), intent(in)              :: Emev                         !< The energy at which to evaluate the cross section in MeV
+    real(8), intent(in)              :: sqrtkT                       !< The temperature in the form sqrt(kT (in eV)), at which to evaluate the cross section.
+    real(8), intent(out)             :: sigT                         !< Total cross section
+    real(8), intent(out)             :: sigA                         !< Absorption cross section
+    real(8), intent(out)             :: sigF                         !< Fission cross section
+    complex(8) :: PSIIKI
+    complex(8) :: CDUM1
+    complex(8) :: w_val
+    complex(8) :: Z
+    real(8) :: sqrtE
+    real(8) :: invE
+    real(8) :: DOPP
+    real(8) :: DOPP_ECOEF
+    real(8) :: temp
+    real(8) :: E
+    integer :: iP
+    integer :: iC
+    integer :: iW
+    integer :: startw
+    integer :: startw_1
+    integer :: startw_endw
+    integer :: endw
+
+    ! Convert to eV
+    E = Emev * 1.0D6
+
+    sqrtE = sqrt(E)
+    invE = E**(-1)
```

Comment:
I actually don't know if 1/E is faster or not.  I'm pretty sure there is some highly optimized algorithm to perform inverse that doesn't trigger for the mathematically equivalent division.  Either way, clarity prevails, as the sqrt immediately above it is far more expensive anyways.